lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -204,6 +204,11 @@ def deploy_has_priority_resources?(resources)
     end
 
     def predeploy_priority_resources(resource_list)
+      bare_pods = resource_list.select { |resource| resource.is_a?(Pod) }
+      if bare_pods.count == 1
```

Comment:
I'm on the fence between only doing this if 1 pod is present vs. picking the first if there are multiple.

Reply:
@KnVerey Any thoughts?

Reply:
I think the UX of picking a random one might be kinda weird--would people think the others aren't actually running? Might be especially weird if we happen to pick one that runs quickly, so we stream for like 2s, and then there are others that take minutes, where there's no output--people might think we've hung. 

This might have weird UX too, but another option could be to stream logs whenever we find there is only one pod (either to begin with, or one pod left) in the set by setting `stream_logs` much like this but from the loop in `ResourceWatcher`.

Reply:
Alternatively, we stream the logs from all pods and interleave them with the pod name as a prefix?

Reply:
> Alternatively, we stream the logs from all pods and interleave them with the pod name as a prefix, perhaps even allowing the user to disable log streaming via an annotation?

We actually discussed the multiple streaming w/ interleaving on the PR that introduced the ability to stream. In the end it was rather ugly & not very helpful.

Reply:
I’ll happily implement whatever y’all will accept. Just let me know.

Reply:
>I'm on the fence between only doing this if 1 pod is present vs. picking the first if there are multiple.

My vote for now is to do this when only 1 pod is present (current behaviour)

Reply:
`bare_pods.count == 1` (as is) seems to be the most popular choice, so let's keep it as is. I think all we need now is a bit of testing.

Reply:
Cool. I'll come back to this once https://github.com/Shopify/kubernetes-deploy/pull/439 is in.

Reply:
Tests added.